src/Elastic.Apm/BackendComm/BackendCommUtils.cs

```diff
@@ -21,6 +21,58 @@ internal class BackendCommUtils
 
 		private static readonly TimeSpan DnsTimeout = TimeSpan.FromMinutes(1);
 
+		internal static class ApmServerEndpoints
+		{
+			internal static Uri BuildIntakeV2EventsAbsoluteUrl(Uri baseUrl) =>
+				CombineAbsoluteAndRelativeUrls(baseUrl, "intake/v2/events");
+
+			internal static Uri BuildGetConfigAbsoluteUrl(Uri baseUrl, Service service)
```

Comment:
I guess this is the fix itself, maybe I'd add some comments here, something like:

```
/// <summary>
/// Builds the absolute URL that points to the server URL that gives the configuration to the agent with the specific service.environment
/// </summary>
/// <param name="baseUrl"></param>
/// <param name="service">Points to the service which will be URL encoded in the returned URL</param>
/// <returns></returns>
```

Just saying because it took me a while to figure out what's going on here.